plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/BooleanCoercer.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.coercions;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.trino.spi.TrinoException;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.type.BooleanType;
+import io.trino.spi.type.VarcharType;
+
+import static io.airlift.slice.SliceUtf8.countCodePoints;
+import static io.trino.spi.StandardErrorCode.INVALID_ARGUMENTS;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+
+public class BooleanCoercer
+{
+    private BooleanCoercer() {}
+
+    public static class BooleanToVarcharCoercer
+            extends TypeCoercer<BooleanType, VarcharType>
+    {
+        public BooleanToVarcharCoercer(VarcharType toType)
+        {
+            super(BOOLEAN, toType);
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            boolean value = BOOLEAN.getBoolean(block, position);
+            Slice converted = Slices.utf8Slice(Boolean.toString(value).toUpperCase(ENGLISH));
```

Comment:
These casts (boolean → string → slice) look redundant to me. The value takes only "true" or "false".